// @next-auth/typeorm-legacy-adapter/dist/models/user.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.UserSchema = exports.User = void 0;

class User {
  constructor(profile) {
    this.name = profile === null || profile === void 0 ? void 0 : profile.name;
    this.email = profile === null || profile === void 0 ? void 0 : profile.email;
    this.image = profile === null || profile === void 0 ? void 0 : profile.image;
    this.emailVerified = (profile === null || profile === void 0 ? void 0 : profile.emailVerified) && new Date();
  }

}

exports.User = User;
const UserSchema = {
  name: "User",
  target: User,
  columns: {
    id: {
      primary: true,
      type: "int",
      generated: true
    },
    name: {
      type: "varchar",
      nullable: true
    },
    email: {
      type: "varchar",
      unique: true,
      nullable: true
    },
    emailVerified: {
      type: "timestamp",
      nullable: true
    },
    image: {
      type: "varchar",
      nullable: true
    },
    createdAt: {
      type: "timestamp",
      createDate: true
    },
    updatedAt: {
      type: "timestamp",
      updateDate: true
    }
  }
};
exports.UserSchema = UserSchema;
